addons/knowledge/models/knowledge_article.py

```diff
@@ -525,7 +525,8 @@ def action_set_unlock(self):
             article.is_locked = False
 
     def action_archive(self):
-        return super(Article, self | self._get_descendants()).action_archive()
+        super(Article, self | self._get_descendants()).action_archive()
+        return self.action_home_page()
```

Comment:
If you do this, when you archive an article from the list view, this will open the home page.. which is not really convenient.
You need to do it in js.